### PR TITLE
fix read & write example in README

### PR DESCRIPTION
**Pull request by dingiso**
I have tried the example in REAMDE. I think several lines should be fixed.
1. correct the head file in read example because of the STFInstReader we use 
2. add a space between pc and opcode 
3. the inst reader will user vlen so it should be set in write example 

And maybe we can add a CMake file example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ stf::STFWriter stf_writer;
 stf_writer.open("my_trace.zstf");  // If you do not want compression, drop the 'z': my_trace.stf
 stf_writer.addTraceInfo(stf::TraceInfoRecord(stf::STF_GEN::STF_GEN_IMPERAS, 1, 2, 0, "Trace from Imperas"));
 stf_writer.setISA(stf::ISA::RISCV);
+stf_writer.setVLen( /*VLEN*/ );
```
Comment (bdutro) [inline README.md:76]: The VLen is actually optional. Maybe add a comment explaining this?

**Review by bdutro — CHANGES_REQUESTED** [review]

**Comment by dingiso** [comment]
Hi bdutro, An assert came out when I try to use the read example to read a stf file, 

```shell
terminate called after throwing an instance of 'stf::STFException'
  what():  vlen >= MIN_VLEN: Vlen parameter (0) must be greater than or equal to 64: in file: '**/stf_lib/lib/stf_fstream.cpp', on line: 28
[1]    3969317 abort (core dumped)  ./stf_read ./new.stf
```
And this is part of the coredump backtrace 

```
#6  0x000055aade4b258f in stf::STFFstream::setVLen(unsigned int) [clone .cold] ()
#7  0x000055aade52358b in stf::RecordFactoryRegistrations::VLenConfigRecord_FactoryMethod(stf::STFIFstream&) [clone .lto_priv.0] ()
```
I dig into the code and find it maybe happens in the `REGISTER_RECORD` macro for VLenConfig and the function call hierachy is 

```
REGISTER_RECORD() -> VLenConfigRecord_FactoryMethod() -> construct -> construct_ -> unpack_impl -> setVlen 
```
In unpack_impl() , read will set vlen to 0 since stf do not have a vlen config and trigger the stf_assert.

```cpp
__attribute__((always_inline))
inline void unpack_impl(STFIFstream& reader) {
    GenericSingleDataRecord::unpack_impl(reader);
     reader.setVLen(getVLen());
}
// getVLen()
vlen_t getVLen() const { return GenericSingleDataRecord::getData_(); }
// setVLen()
void STFFstream::setVLen(const vlen_t vlen) {
    static constexpr size_t MIN_VLEN = byte_utils::bitSize<InstRegRecord::ValueType>();
    stf_assert(!vlen_, "Attempted to set VLen multiple times");
    stf_assert(vlen >= MIN_VLEN,
                "Vlen parameter ("
                << vlen
                << ") must be greater than or equal to "
                << MIN_VLEN);
    vlen_ = vlen;
}
```
If the VLen is optional, I think this can be fixed by change the assert to warn or add a check in unpack_impl

**Comment by dingiso** [comment]
Thank @bdutro  for your explaination. I get the point and correct the pr with comment for func `setVLen`

File: README.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ stf::STFInstReader reader(trace_filename,  // *.zstf, *.stf
 
 // Iterate through the trace
 for(const stf::STFInst & inst : reader) {
-    std::cout << std::hex << "PC: 0x" << inst.pc() << "Opcode: 0x" << inst.getOpcode() << std::endl;
+    std::cout << std::hex << "PC: 0x" << inst.pc() << " Opcode: 0x" << inst.getOpcode() << std::endl;
```
Comment (knute-sifive) [inline README.md:54]: getOpcode needs to be `opcode`

Comment (bdutro) [reply README.md:54]: I'll fix this separately since I can't edit the PR.

**Review by bdutro — APPROVED** [review]

**Pull request merged**